compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -2105,51 +2105,52 @@ static LogicalResult setVMVXRootConfigImpl(func::FuncOp entryPointFn,
-  auto linalgOp = dyn_cast<linalg::LinalgOp>(rootOp);
-  if (!linalgOp)
-    return success();
-
-  auto loweringConfig = getLoweringConfig(linalgOp);
-  TileSizesListType tileSizesList = loweringConfig.getTileSizeVals();
-
-  bool hasChanged = false;
-  auto res = entryPointFn.walk([&](tensor::PackOp packOp) -> WalkResult {
-    // Multiple pack ops case is not supported.
-    if (hasChanged)
-      return WalkResult::interrupt();
-
-    hasChanged = true;
-    LLVM_DEBUG(KD_DBGS() << "Find pack op candidate: " << packOp << "\n");
-
-    // Only adjust tile sizes for distribution and TileAndFuse, which are the
-    // first two tile lists.
-    // Align the tile sizes of the root op to the pack op's inner tile sizes, so
-    // we can derive the outer tile sizes for pack ops later in
-    // setLoweringConfigForComputeOps by dividing with inner tile sizes.
-    for (int i = 0, e = std::min<int>(tileSizesList.size(), 2); i < e; ++i) {
-      auto &tileSizes = tileSizesList[i];
-      ArrayRef<int64_t> innerTiles = packOp.getStaticInnerTiles();
-      ArrayRef<int64_t> dimPos = packOp.getInnerDimsPos();
-      for (auto [pos, size] : llvm::zip_equal(dimPos, innerTiles)) {
-        if (tileSizes[pos] == 0 || ShapedType::isDynamic(size))
-          continue;
-        tileSizes[pos] = llvm::alignTo(tileSizes[pos], size);
-        LLVM_DEBUG(KD_DBGS() << "Align # " << pos << " tile size to "
-                             << tileSizes[pos] << "\n");
-      }
-    }
-
-    return WalkResult::advance();
-  });
-  if (res.wasInterrupted())
+static LogicalResult
+adjustTileSizesForPackOp(func::FuncOp entryPointFn, tensor::PackOp packOp,
+                         SmallVector<int64_t> &distTileSizes,
+                         SmallVector<int64_t> &parallelVecTileSizes) {
+  // Align the tile sizes of the root op to the pack op's inner tile sizes, so
+  // we can derive the outer tile sizes for pack ops later in
+  // setLoweringConfigForComputeOps by dividing with inner tile sizes.
+  ArrayRef<int64_t> innerDimsPos = packOp.getInnerDimsPos();
+  ArrayRef<int64_t> innerTiles = packOp.getStaticInnerTiles();
+  // Currently we only handle pack op with static inner tile sizes.
+  if (llvm::any_of(innerTiles,
+                   [](int64_t size) { return ShapedType::isDynamic(size); })) {
     return failure();
-
-  auto pipeline = getTranslationInfo(entryPointFn).getPassPipeline().getValue();
-  return setOpConfigAndEntryPointFnTranslation(
-      entryPointFn, rootOp, tileSizesList,
-      loweringConfig.getScalableTileFlagVals(), pipeline);
+  }
+  // Pack op requires special vector tile sizes to achieve good performance.
+  // Override the parallel vector tile sizes from pack op.
+  auto vecTileSizes = getPackVectorTileSizes(entryPointFn, packOp);
+  auto outerDimsPerm = packOp.getOuterDimsPerm();
+  if (!outerDimsPerm.empty()) {
+    auto invertedPerm = invertPermutationVector(outerDimsPerm);
+    applyPermutationToVector(vecTileSizes, invertedPerm);
+  }
```

Comment:
The `getPackVectorTilSizes` returns the tiling config for the pack op, which is permuted of its input dimensions. So to get the tile sizes for other ops, we need to invert-permute the tile sizes from `getPackVectorTilSizes`.

Later we permute the joined tile size for the pack op:
https://github.com/openxla/iree/blob/8b1e2accc28c6173d9577c49c20b0d5817e591da/compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp#L2306-L2321

Therefore I don't think the permutation belongs to `getPackVectorTilSizes` because it returns the tile sizes for the pack op. The permutation here is to transform the tile sizes from pack op to other ops

Reply:
oh I see, good point!